lib/bitstyles_phoenix/component/form.ex

```diff
@@ -0,0 +1,555 @@
+      ...> render ~H"""
+      ...> <.ui_select form={@form} field={:week} options={1..2} />
+      ...> """
+      """
+      <label for="user_week">
+        Week
+      </label>
+      <select id="user_week" name="user[week]">
+        <option value="1">
+          1
+        </option>
+        <option value="2">
+          2
+        </option>
+      </select>
+      """
+  ''')
+
+  story("Select box without label", '''
+      iex> assigns=%{form: @user_form}
+      ...> render ~H"""
+      ...> <.ui_select form={@form} field={:week} options={1..2} hidden_label/>
+      ...> """
+      """
+      <label class="u-sr-only" for="user_week">
+        Week
+      </label>
+      <select id="user_week" name="user[week]">
+        <option value="1">
+          1
+        </option>
+        <option value="2">
+          2
+        </option>
+      </select>
+      """
+  ''')
+
+  story("Select box with options", '''
+      iex> assigns=%{form: @user_form, options: [{"Ducks", "ducks"}, {"Cats", "cats"}]}
+      ...> render ~H"""
+      ...> <.ui_select form={@form} field={:preference} options={@options} label="What do you like best?" label_opts={[class: "extra"]}/>
+      ...> """
+      """
+      <label class="extra" for="user_preference">
+        What do you like best?
+      </label>
+      <select id="user_preference" name="user[preference]">
+        <option value="ducks">
+          Ducks
+        </option>
+        <option value="cats">
+          Cats
+        </option>
+      </select>
+      """
+  ''')
+
+  def ui_select(assigns) do
+    extra = assigns_to_attributes(assigns, @wrapper_assigns_keys ++ [:options])
+
+    assigns =
+      assigns
+      |> assign(:extra, extra)
+      |> assign(:wrapper, wrapper_assigns(assigns))
+
+    ~H"""
+    <.ui_unwrapped_input {@wrapper}>
+      <%= PhxForm.select(@form, @field, @options, @extra) %>
+    </.ui_unwrapped_input>
+    """
+  end
+
+  defp wrapper_assigns(assigns) do
+    Map.take(assigns, @wrapper_assigns_keys)
+  end
+
+  @doc """
+  Component for rendering custom inputs together with a label and errors.
+
+  ## Attributes
+
+  - `hidden_label` - Only show the label for screen readers if set to `true`.
+  - All options from above (see top level module doc).
+  """
+
+  story("Custom inputs", '''
+      iex> assigns=%{form: @error_form}
+      ...> render ~H"""
+      ...> <.ui_unwrapped_input form={@form} field={:name} label="Custom">
+      ...>   Custom content
+      ...>   <input type="text" whatever="foo" />
+      ...> </.ui_unwrapped_input>
+      ...> """
+      """
+      <label for="user_name">
+        Custom
+      </label>
+      Custom content
+      <input type="text" whatever="foo"/>
+      <span class="u-fg--warning" phx-feedback-for="user[name]">
+        is too short
+      </span>
+      """
+  ''')
+
+  def ui_unwrapped_input(assigns) do
+    label_text = Map.get_lazy(assigns, :label, fn -> default_label(assigns.field) end)
```

Comment:
```
assigns =
  assigns
  |> assign_new(:label, fn -> default_label(assigns.field) end)
```

reads better IMO

Reply:
But it does something else ...

Reply:
🤔  hmm ok I see now, you're actually overwriting the `label` assign with the `PhxForm.label/4` result... How about calling the option `label_text` instead and turning label into a slot or so?